src/main/kotlin/at/bitfire/dav4jvm/DavResource.kt

```diff
@@ -722,9 +722,29 @@ open class DavResource @JvmOverloads constructor(
         return response
     }
 
+    /**
+     * Checks whether the body of a [response] is set or not. This is done by checking the `Content-Length` header.
+     * - If it's `0`, and no `Content-Type` is set, no body is given.
+     * - If it's `-1` (not set), and `Transfer-Encoding` is not `chunked`, no body is given.
+     * Note that this last condition is necessary because chunked body requests doesn't have a `Content-Length`.
+     *
+     * This request will call [Response.body], but won't close it. It's responsibility of the caller to properly close it.
+     * @return `true` if [response] has a valid body, `false` otherwise.
+     */
+    private fun isEmptyResponseBody(response: Response): Boolean {
```

Comment:
The name indicates the opposite as what the method does:

- name: is empty response body
- return value: is valid response body

However, I think we can completely drop this method: as [response.body is now guaranteed to be non-null](https://github.com/square/okhttp/blob/d5b3a7471baf290e9040e6d579fa0844d23427bf/CHANGELOG.md?plain=1#L479-L482), we don't need the `?: throw Exception("empty response body")` anymore.

We can just process the body, and if we get an empty XML element, parsing will fail. We may however add a test to see how `processMultiStatus()` reacts to an empty body, and make sure that we get a useful exception in that case.